modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -73,6 +80,53 @@ const AddFDSFieldsModalContent = ({
 		);
 	};
 
+	const synchronizeFDSFields = async () => {
+		const creationData: Array<{name: string; type: string}> = [];
+		const deletionIds: Array<number> = [];
+
+		fields?.forEach((field) => {
+			if (field.selected && !field.id) {
+				creationData.push({name: field.name, type: field.type});
+			}
+
+			if (!field.selected && field.id) {
+				deletionIds.push(field.id);
+			}
+		});
+
+		const formData = new FormData();
+
+		formData.append(
+			`${namespace}creationData`,
+			JSON.stringify(creationData)
+		);
+
+		deletionIds.forEach((id) => {
+			formData.append(`${namespace}deletionIds`, String(id));
+		});
+
+		formData.append(`${namespace}fdsViewId`, fdsView.id);
+
+		const response = await fetch(synchronizeFDSFieldsURL, {
+			body: formData,
+			method: 'POST',
+		});
+
+		if (response.ok) {
+			closeModal();
+
+			onSave();
+		}
+		else {
+			openToast({
```

Comment:
Hmm we could return IDs of newly created items, and apply them as fetched. With that, we can drop the `getFields` request after save. It's worth doing, as a part of this PR, or a followup.

Reply:
We may even consider lifting the state of FDSView to the parent component and then sync all data from there, could easily do this with react context

Reply:
>We may even consider lifting the state of FDSView to the parent component and then sync all data from there, could easily do this with react context

I was very close at one point to setting props of `FDSViewSectionInterface` as context props. I'd say this would be justified if we had three or more levels of hierarchy. I don't think we currently do, it's FDS View > tab content. This might easily change. 

It may make sense to hoist state of child component (fields, filters etc) to `FDSView`, at some point. State should be in the lowest scope it affects, that's objectively unshakable rule achieving best performance and separation of concerns. We should not shift it up just because we can. But, if for some reason state in child component affects state in parent or sibling, rewrite is in order.